src/sys.zig

```diff
@@ -142,13 +148,159 @@ pub const Tag = enum(u8) {
 
     pub var strings = std.EnumMap(Tag, JSC.C.JSStringRef).initFull(null);
 };
-const PathString = @import("root").bun.PathString;
 
-const mode_t = os.mode_t;
+pub const Error = struct {
+    const E = bun.C.E;
 
-const open_sym = system.open;
+    const retry_errno = if (Environment.isLinux)
+        @as(Int, @intCast(@intFromEnum(E.AGAIN)))
+    else if (Environment.isMac)
+        @as(Int, @intCast(@intFromEnum(E.WOULDBLOCK)))
+    else
+        @as(Int, @intCast(@intFromEnum(E.INTR)));
 
-const mem = std.mem;
+    const todo_errno = std.math.maxInt(Int) - 1;
+
+    pub const Int = if (Environment.isWindows) u16 else u8; // @TypeOf(@intFromEnum(E.BADF));
+
+    /// TODO: convert to function
+    pub const oom = fromCode(E.NOMEM, .read);
+
+    errno: Int = todo_errno,
```

Comment:
Using `todo_errno` as the default `errno` value allows for specifying errors without pre-populating it manually.